moveit_core/online_signal_smoothing/CMakeLists.txt

```diff
@@ -28,16 +28,26 @@ target_include_directories(moveit_butterworth_filter PUBLIC
 set_target_properties(moveit_butterworth_filter PROPERTIES VERSION
   "${${PROJECT_NAME}_VERSION}"
 )
+
+generate_parameter_library(butterworth_parameters src/butterworth_parameters.yaml)
+
 target_link_libraries(moveit_butterworth_filter
   moveit_smoothing_base
   moveit_robot_model
+  butterworth_parameters
 )
 ament_target_dependencies(moveit_butterworth_filter
   srdfdom  # include dependency from moveit_robot_model
 )
 
 # Installation
-
+install(
+  TARGETS butterworth_parameters
+  EXPORT ${PROJECT_NAME}Targets
```

Comment:
I hope @tylerjw or @ChrisThrasher will stop by to say if this was done correctly. It's a little different from what was done here, for example:

https://github.com/ros-controls/ros2_controllers/blob/fa7499f0eb28ccca11408e703b8d5fce4f4deb30/joint_trajectory_controller/CMakeLists.txt#L97

Reply:
Would be great to get some feedback on this point.
The reference I followed : https://github.com/ros-planning/moveit2/blob/9d6ceb20d3f9046b287a1e4540af8aa73199e2e0/moveit_kinematics/CMakeLists.txt#L74

Reply:
This is probably not doing what you want. To make CMake targets visible to downstream packages there are a few things you need to do of which you're doing one of them. The first step is creating an export set which is what `install(TARGETS ... EXPORT` is doing. `${PROJECT_NAME}Targets` is your export set. This is a collection of CMake targets that you want to install. Alongside creating an export set you specify where various binaries from that export set get installed which you're already doing.

You also have to install that export set via `install(EXPORT` then create a config module which includes the file that `install(EXPORT` generates. I'm guessing that this machinery already exists elsewhere in moveit_core and that you may jus need to list this target in some other export set to automatically get that behavior.

I'm not 100% sure how moveit_core architectures its install nor am I 100% how the ament CMake wrappers ~~obscure~~ simplify this so you may not need to explicitly write those other two things I mentioned if ament is automagically doing that for you.

---

As an aside, please don't use `${PROJECT_NAME}` here though. It puts extra burden on those reading the code and makes it much harder to `grep` for things wherever it is used. It's analogous to using a C++ macro to specify a variable name instead simply spelling out that variable name each time you need it.